lib/MockXMLHttpRequest.js

```diff
@@ -158,40 +159,45 @@ MockXMLHttpRequest.prototype.send = function(data) {
 
     var response = MockXMLHttpRequest.handle(new MockRequest(self));
 
-    if (response && response instanceof MockResponse) {
+    Promise.resolve(response).then(function(response) {
```

Comment:
@jameslnewell Interesting point but i'm not sure that we need this.

Let's look on this:

``` javascript
  /**
   * Mock a request
   * @param   {string}    [method]
   * @param   {string}    [url]
   * @param   {Function}  fn
   * @returns {exports}
   */
  mock: function(method, url, fn) {
    var handler, matcher;
    if (arguments.length === 3) {
      function match(string, pattern) {
        return new RegExp(pattern).test(string);
      }
      handler = function(req, res) {
        if (match(req.method(), method)
            && match(req.url(), url)) {
          return fn(req, res);
        }
        return false;
      };
    } else {
      handler = method;
    }

    MockXMLHttpRequest.addHandler(handler);
  },
```

We already return **false** in case if request doesn't match. This doesn't matter what is Response (MockResponse or Promise) and our `send()` should handle this correctly, trigger **error** event.

Another case, we have mapping for ANY response with just handler as you mentioned above

```javascript
function handler(req, res) {
  return new Promise(function(resolve, reject) {
    //resolving without a response if we decided to not handle the request
    setTimeout(resolve, 100);
  });
}
```

Yes it will return no body and this will trigger **error** event as well, what is correct for me.

But in my opinion is not correct if XHR MOCK will try to return response from another request handler that was able fulfill it, it's weird.

I think Developer who use this tool should be careful when specifies mocked requests

What if he has mocked several requests for ANY requests but one responses with timeout and another returns immediately. Which one should response to him? First or second? Or maybe developer should be accurate and clean old mocks?